### PR TITLE
Update deprecated `&x` syntax  in blas

### PR DESCRIPTION
**Pull request by musm**
part of #6080

### CODE REVIEW

**Comment by fredrikekre** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/23400#issuecomment-324170692):
```
NanosoldierError: failed to run benchmarks against primary commit: failed process: Process(`sudo cset shield -e su nanosoldier -- -c ./benchscript.sh`, ProcessExited(1)) [1]
```
Logs and partial data can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/5ca85b781351108ac5edf9f2212ecd9f6f92bad3/9cdde59_vs_49c44de/report.md)
cc @ararslan

**Comment by ararslan** [comment]
Hm, that's weird, let's try again. @nanosoldier `runbenchmarks(ALL, vs=":master")`

File: base/linalg/blas.jl [hunk 1]
```diff
@@ -1312,12 +1311,12 @@ for (fname, elty, relty) in ((:zherk_, :Complex128, :Float64),
...
-                  Ptr{$relty}, Ptr{$elty}, Ptr{BlasInt}, Ptr{$relty},
-                  Ptr{$elty}, Ptr{BlasInt}),
-                 &uplo, &trans, &n, &k,
-                 &α, A, &max(1,stride(A,2)), &β,
-                 C, &max(1,stride(C,2)))
+                 (Ref{UInt8}, Ref{UInt8}, Ref{BlasInt}, Ref{BlasInt},
+                  Ref{$relty}, Ref{$elty}, Ref{BlasInt}, Ref{$relty},
+                  Ref{$elty}, Ref{BlasInt}),
+                  uplo, trans, n, k,
+                  α, A, max(1,stride(A,2)), &β,
```
Comment (fredrikekre) [inline base/linalg/blas.jl]: Missed `&` here? Nanosoldier logs complained about this so probably the reason for the benchmark failure.

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/23400#issuecomment-324176359):
```
NanosoldierError: failed to run benchmarks against primary commit: failed process: Process(`sudo cset shield -e su nanosoldier -- -c ./benchscript.sh`, ProcessExited(1)) [1]
```
Logs and partial data can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/23363e023fa95e21e108deab7edf3d9fd488d571/9cdde59_vs_49c44de/report.md)
cc @ararslan

**Comment by musm** [comment]
one more time?

**Comment by ararslan** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23400#issuecomment-324209388) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/7769087a127a50685d05e753000b7ad3b1409070/bacf1d0_vs_086eca1/report.md). cc @ararslan

**Comment by andreasnoack** [comment]
I can confirm that this actually increases the memory usage in the the `["random"]["types"][("rand", "MersenneTwister", "UInt32")]` benchmark (but not the number of allocations). @yuyichao Is that expected?

File: base/linalg/blas.jl [hunk 2]
```diff
@@ -171,8 +171,8 @@ for (fname, elty) in ((:dcopy_,:Float64),
         # SUBROUTINE DCOPY(N,DX,INCX,DY,INCY)
         function blascopy!(n::Integer, DX::Union{Ptr{$elty},StridedArray{$elty}}, incx::Integer, DY::Union{Ptr{$elty},StridedArray{$elty}}, incy::Integer)
             ccall((@blasfunc($fname), libblas), Void,
-                (Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt}),
```
Comment (Keno) [inline base/linalg/blas.jl]: Here and in a few other places, there's no ampersand on the `DX` and `DY`. Probably still fine by proper conversions. Just wanted to point it out.

Comment (musm) [inline base/linalg/blas.jl]: I was going off the docs: https://github.com/JuliaLang/julia/pull/23307#discussion_r133811522

Comment (musm) [inline base/linalg/blas.jl]: I can revert this back if necessary, but it seems that they should in principle be Refs?

**Comment by andreasnoack** [comment]
@musm Why close?

**Comment by musm** [comment]
@Keno mentioned he has a bot that can fix these, so I thought he could use this and lapack.jl with the bot.

**Pull request closed without merging**